### PR TITLE
Full multijob matrix gha

### PR DESCRIPTION
**Pull request by poshul**
## Description

Adds full building of our packages as a github action

TODO
- Deactivate/remove old action
- Test package and knime
- Add CTest script for Knime (like for build test package)
- Make sure they are uploaded to correct place in Cdash
- Test uploads
- Add ssh secrets
- Test ssh

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by poshul** [comment]
@jpfeuffer  whats up with: DO_PACKAGE=$( [[ ( "${{ inputs.package }}" || "${{ steps.extract_branch.outputs.BASE_REF }}" == "nightly" || \ ...  ? Github keeps throwing an error that the whole `run` block evaluates to a bool?

File: .github/workflows/openms_ci_matrix_full.yml [hunk 1]
```diff
@@ -0,0 +1,565 @@
+# This is a basic workflow to help you get started with Actions
```
Comment (timosachsenberg) [inline .github/workflows/openms_ci_matrix_full.yml]: document a bit what it does.
e.g., what happens if someone opens a PR against develop?
does this also build packages? where is stuff uploaded to? etc.

**Comment by jpfeuffer** [comment]
Just some usual bash syntax insanity.

**Comment by jpfeuffer** [comment]
Delete openms-ci or at least it's triggers first?

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**